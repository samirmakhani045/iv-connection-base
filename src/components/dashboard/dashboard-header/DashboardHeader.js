import React from 'react';
import PropTypes from 'prop-types';

import { useTranslation } from '@studio/hooks/useTranslation';

import { Wrapper, H1, StyledButton } from './DashboardHeader.styles';

const DashboardHeader = (props) => {
  const { title, onCreateFlow } = props;
  const { t } = useTranslation('pages.dashboard');

  return (
    <Wrapper data-testid="wrapper">
      <H1>{title}</H1>

      {onCreateFlow && (
        <StyledButton
          label={t('menu.create')}
          icon="Add"
          variant="light"
          onClick={onCreateFlow}
          small
        />
      )}
    </Wrapper>
  );
};

DashboardHeader.propTypes = {
  /** Title to give context to user on dashboard */
  title: PropTypes.string,
  /** Callback to create flow. Hide create flow button if undefined */
  onCreateFlow: PropTypes.func,
};

export default DashboardHeader;
