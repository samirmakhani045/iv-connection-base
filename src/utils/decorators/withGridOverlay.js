import React from 'react';
import styled from 'styled-components/macro';

import * as colors from '@studio/theme/colors';
import { Grid, Box } from '@studio/components/grid';

const Div = styled.div({
  position: 'relative',
});

const StyledGrid = styled(Grid)({
  height: 'calc(100vh - 2rem)',
  left: '50%',
  pointerEvents: 'none',
  position: 'absolute',
  top: '50vh',
  transform: 'translate(-50%, -50vh)',
  width: '100%',
  zIndex: '1000',
});

const StyledBox = styled(Box)({
  backgroundColor: colors.red,
  height: '100%',
  opacity: '0.1',

  '&:nth-child(odd)': {
    opacity: '0.15',
  },
});

const withGridOverlay = (story, props) => {
  if (!props?.args?.gridOverlay) return <>{story()}</>;

  return (
    <Div>
      {story()}
      <StyledGrid>
        <StyledBox />
        <StyledBox />
        <StyledBox />
        <StyledBox />
        <StyledBox />
        <StyledBox />
        <StyledBox />
        <StyledBox />
        <StyledBox />
        <StyledBox />
        <StyledBox />
        <StyledBox />
        <StyledBox />
        <StyledBox />
        <StyledBox />
        <StyledBox />
      </StyledGrid>
    </Div>
  );
};

export default withGridOverlay;
