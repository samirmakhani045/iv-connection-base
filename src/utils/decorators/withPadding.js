import React from 'react';
import styled from 'styled-components/macro';

const Wrapper = styled.div({
  padding: '3rem',
});

const withPadding = (story) => <Wrapper>{story()}</Wrapper>;

export default withPadding;
