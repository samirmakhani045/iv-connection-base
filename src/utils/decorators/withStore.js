import React from 'react';
import { Provider } from 'react-redux';

import configureStore from '@studio/store/configureStore';

const withStore = (story) => {
  const store = configureStore();

  return <Provider store={store}>{story()}</Provider>;
};

export default withStore;
