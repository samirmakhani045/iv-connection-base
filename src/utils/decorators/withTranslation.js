import React from 'react';
import { Provider } from 'react-redux';
import { I18nextProvider } from 'react-i18next';

import i18n from '@studio/i18n';
import localTranslations from '@studio/translations';
import configureStore from '@studio/store/configureStore';
import { flatten } from '@utils/flatten';

const withTranslation = (story) => {
  const store = configureStore();
  i18n.addResources('en', 'translations', flatten(localTranslations));

  return (
    <I18nextProvider i18n={i18n}>
      <Provider store={store}>{story()}</Provider>
    </I18nextProvider>
  );
};

export default withTranslation;
