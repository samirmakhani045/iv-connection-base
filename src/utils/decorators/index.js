import withGridOverlay from './withGridOverlay';
import withPadding from './withPadding';
import withStore from './withStore';
import withTranslation from './withTranslation';

export { withGridOverlay, withPadding, withStore, withTranslation };
