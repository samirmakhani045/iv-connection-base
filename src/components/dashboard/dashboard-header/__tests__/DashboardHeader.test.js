import React from 'react';
import { render, screen } from '@testing-library/react';

import { withTranslation } from '@utils/decorators';

import DashboardHeader from '../DashboardHeader';

const defaultProps = {
  title: 'Title',
};

const Test = (props) => withTranslation(() => <DashboardHeader {...defaultProps} {...props} />);

describe('<DashboardHeader />', () => {
  it('should render correctly', () => {
    const { container } = render(<Test />);

    expect(screen.queryByTestId('wrapper')).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
