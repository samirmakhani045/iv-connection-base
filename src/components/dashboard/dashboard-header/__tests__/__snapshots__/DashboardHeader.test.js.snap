// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DashboardHeader /> should render correctly 1`] = `
<div
  class="DashboardHeaderstyles__Wrapper-sc-1in23sz-0 cfrswK"
  data-testid="wrapper"
>
  <h1
    class="DashboardHeaderstyles__H1-sc-1in23sz-1 gYxHqX"
  >
    Title
  </h1>
</div>
`;
